File: prm393_g3_backend/src/models/variant.js
import mongoose from 'mongoose';

const variantSchema = new mongoose.Schema(
  {
    productId: { type: mongoose.Schema.Types.ObjectId, ref: 'product', required: true },
    sku: { type: String, required: true, unique: true },
    barcode: { type: String, unique: true },
    price: { type: Number, required: true },
    status: { type: String, enum: ['ACTIVE', 'INACTIVE'], default: 'ACTIVE' },
  },
  { timestamps: true },
);

// Explicitly map to 'variants' collection
const Variant = mongoose.model('variant', variantSchema, 'variants');

export default Variant;

File: prm393_g3_frontend/lib/features/variants/domain/entities/variant.dart
import 'package:equatable/equatable.dart';

class Variant extends Equatable {
  const Variant({
    required this.id,
    required this.productId,
    required this.sku,
    this.barcode,
    required this.price,
    required this.status,
    this.variantName,
  });

  final String id;
  final String productId;
  final String sku;
  final String? barcode;
  final num price;
  final String status; // 'ACTIVE' or 'INACTIVE'
  final String? variantName;

  @override
  List<Object?> get props => [id, productId, sku, barcode, price, status, variantName];
}
